Core/include/Acts/Utilities/Intersection.hpp

```diff
@@ -39,145 +39,227 @@ inline std::ostream& operator<<(std::ostream& os, IntersectionStatus status) {
-      : intersection(sInter),
-        object(sObject),
-        representation(sRepresentation) {}
-
-  /// Bool() operator for validity checking
-  explicit operator bool() const { return bool(intersection); }
-
-  /// @brief Smaller operator for ordering & sorting
-  ///
-  /// This operator will ignore the alternative, but simply
-  /// order the representing intersection
-  ///
-  /// @param oi is the source intersection for comparison
-  bool operator<(
-      const ObjectIntersection<object_t, representation_t>& oi) const {
-    return (intersection < oi.intersection);
+  /// @param intersections are the intersections
+  /// @param object is the object to be instersected
+  /// @param representation is the object representation
+  constexpr ObjectMultiIntersection(const MultiIntersection3D& intersections,
+                                    const object_t* object,
+                                    const representation_t* representation)
+      : m_intersections(intersections),
+        m_object(object),
+        m_representation(representation) {}
+
+  constexpr ObjectIntersection<object_t, representation_t> operator[](
+      std::uint8_t index) const {
+    return {m_intersections[index], m_object, m_representation, index};
   }
 
-  /// @brief Greater operator for ordering & sorting
-  ///
-  /// This operator will ignore the alternative, but simply
-  /// order the representing intersection
-  ///
-  /// @param oi is the source intersection for comparison
-  bool operator>(
-      const ObjectIntersection<object_t, representation_t>& oi) const {
-    return (intersection > oi.intersection);
+  constexpr std::size_t size() const { return m_intersections.size(); }
+
+  constexpr const object_t* object() const { return m_object; }
+
+  constexpr const representation_t* representation() const {
+    return m_representation;
   }
 
-  /// Allow swapping the intersection and the alternative
-  void swapSolutions() { std::swap(intersection, alternative); }
-};
+  constexpr std::array<ObjectIntersection<object_t, representation_t>, 2>
+  split() const {
+    return {operator[](0), operator[](1)};
+  }
 
-struct SameSurfaceIntersection {
-  /// @brief comparison operator
-  ///
-  /// This is a struct to pick out intersection with identical surfaces
-  ///
-  /// @tparam intersection_t Type of the intersection object
-  /// @param i1 First intersection to test
-  /// @param i2 Second intersection to test
-  template <typename intersection_t>
-  bool operator()(const intersection_t& i1, const intersection_t& i2) const {
-    return (i1.object == i2.object);
+  constexpr ObjectIntersection<object_t, representation_t> closest() const {
+    auto splitIntersections = split();
+    return *std::min_element(
+        splitIntersections.begin(), splitIntersections.end(),
+        ObjectIntersection<object_t, representation_t>::closestOrder);
   }
+
+ private:
+  /// The intersections
+  MultiIntersection3D m_intersections;
```

Comment:
This typedef could literally be a `static_vector<Intersection3D, 2>`, couldn't it?